worlds/doom_ii/__init__.py

```diff
@@ -0,0 +1,268 @@
+            loc_name = map_name + " - Exit"
+            item_name = map_name + " - Complete"
+
+            if loc_name not in self.location_name_to_id:
+                continue
+
+            if item_name not in self.item_name_to_id:
+                continue
+
+            loc = Locations.location_table[self.location_name_to_id[loc_name]]
+            if not self.included_episodes[loc["episode"] - 1]:
+                continue
+
+            self.multiworld.get_location(loc_name, self.player).place_locked_item(self.create_item(item_name))
+            self.location_count -= 1
+
+        # Give starting levels right away
+        for i in range(len(self.starting_level_for_episode)):
+            if self.included_episodes[i]:
+                self.multiworld.push_precollected(self.create_item(self.starting_level_for_episode[i]))
+        
+        # Give Computer area maps if option selected
+        if getattr(self.multiworld, "start_with_computer_area_maps")[self.player].value:
+            for item_id, item_dict in Items.item_table.items():
+                item_episode = item_dict["episode"]
+                if item_episode > 0:
+                    if item_dict["doom_type"] == DOOM_TYPE_COMPUTER_AREA_MAP and self.included_episodes[item_episode - 1]:
+                        self.multiworld.push_precollected(self.create_item(item_dict["name"]))
+
+        # Fill the rest starting with powerups, then fillers
+        self.create_ratioed_items("Armor", itempool)
+        self.create_ratioed_items("Mega Armor", itempool)
+        self.create_ratioed_items("Berserk", itempool)
+        self.create_ratioed_items("Invulnerability", itempool)
+        self.create_ratioed_items("Partial invisibility", itempool)
+        self.create_ratioed_items("Supercharge", itempool)
+
+        while len(itempool) < self.location_count:
+            itempool.append(self.create_item(self.get_filler_item_name()))
+
+        # add itempool to multiworld
+        self.multiworld.itempool += itempool
+
+    def get_filler_item_name(self):
+        return self.multiworld.random.choice([
+            "Medikit",
+            "Box of bullets",
+            "Box of rockets",
+            "Box of shotgun shells",
+            "Energy cell pack"
+        ])
+
+    def create_ratioed_items(self, item_name: str, itempool: List[DOOM2Item]):
+        remaining_loc = self.location_count - len(itempool)
+        ep_count = self.get_episode_count()
+
+        # Was balanced based on DOOM 1993's first 3 episodes
+        count = min(remaining_loc, max(1, int(round(self.items_ratio[item_name] * ep_count / 3))))
+        if count == 0:
+            logger.warning("Warning, no ", item_name, " will be placed.")
+            return
+
+        for i in range(count):
+            itempool.append(self.create_item(item_name))
+
+    def fill_slot_data(self) -> Dict[str, Any]:
+        return {name: getattr(self.multiworld, name)[self.player].value for name in self.option_definitions}
```

Comment:
Filter this down to only what you need in the client

```suggestion
        return self.options.as_dict("difficulty", "random_monsters", "random_pickups", "random_music", "flip_levels", "allow_death_logic", "pro", "death_link", "reset_level_on_death", "episode1", "episode2", "episode3", "episode4")
```